File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientCpp/src/WeaponUpgradeModelView.cpp
#include "../inc/WeaponUpgradeModelView.h"
void WeaponUpgradeModelView::serialize(std::ostream& output) {
	// Description
	this->Description.serialize(output);
	// PassiveAlterations
	output << this->PassiveAlterations.size() << '\n';
	for(int PassiveAlterations_it = 0; PassiveAlterations_it < this->PassiveAlterations.size(); PassiveAlterations_it++) {
		this->PassiveAlterations[PassiveAlterations_it].serialize(output);
	}

	// Cost
	output << ((float)this->Cost) << '\n';
}

WeaponUpgradeModelView WeaponUpgradeModelView::deserialize(std::istream& input) {
	WeaponUpgradeModelView _obj = WeaponUpgradeModelView();
	// Description
	SpellLevelDescriptionView _obj_Description = SpellLevelDescriptionView::deserialize(input);
	_obj.Description = (::SpellLevelDescriptionView)_obj_Description;
	// PassiveAlterations
	std::vector<StateAlterationModelView> _obj_PassiveAlterations = std::vector<StateAlterationModelView>();
	int _obj_PassiveAlterations_count; input >> _obj_PassiveAlterations_count; input.ignore(1000, '\n');
	for(int _obj_PassiveAlterations_i = 0; _obj_PassiveAlterations_i < _obj_PassiveAlterations_count; _obj_PassiveAlterations_i++) {
		StateAlterationModelView _obj_PassiveAlterations_e = StateAlterationModelView::deserialize(input);
		_obj_PassiveAlterations.push_back((StateAlterationModelView)_obj_PassiveAlterations_e);
	}

	_obj.PassiveAlterations = (::std::vector<StateAlterationModelView>)_obj_PassiveAlterations;
	// Cost
	float _obj_Cost; input >> _obj_Cost; input.ignore(1000, '\n');
	_obj.Cost = (float)_obj_Cost;
	return _obj;
}

WeaponUpgradeModelView::WeaponUpgradeModelView() {
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/WeaponUpgradeModelView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views.Client
{

	public class WeaponUpgradeModelView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		/// <summary>
		/// Obtient du sort que lance l'arme à ce niveau d'upgrade.
		/// </summary>
		public SpellLevelDescriptionView Description;	
		/// <summary>
		/// Obtient les altérations d'état appliquées passivement par l'arme à ce niveau d'upgrade.
		/// </summary>
		public List<StateAlterationModelView> PassiveAlterations;	
		/// <summary>
		/// Obtient le coût de cette upgrade.
		/// </summary>
		public float Cost;	
		public static WeaponUpgradeModelView Deserialize(System.IO.StreamReader input) {
			WeaponUpgradeModelView _obj =  new WeaponUpgradeModelView();
			// Description
			SpellLevelDescriptionView _obj_Description = SpellLevelDescriptionView.Deserialize(input);
			_obj.Description = (SpellLevelDescriptionView)_obj_Description;
			// PassiveAlterations
			List<StateAlterationModelView> _obj_PassiveAlterations = new List<StateAlterationModelView>();
			int _obj_PassiveAlterations_count = Int32.Parse(input.ReadLine());
			for(int _obj_PassiveAlterations_i = 0; _obj_PassiveAlterations_i < _obj_PassiveAlterations_count; _obj_PassiveAlterations_i++) {
				StateAlterationModelView _obj_PassiveAlterations_e = StateAlterationModelView.Deserialize(input);
				_obj_PassiveAlterations.Add((StateAlterationModelView)_obj_PassiveAlterations_e);
			}
			_obj.PassiveAlterations = (List<StateAlterationModelView>)_obj_PassiveAlterations;
			// Cost
			float _obj_Cost = Single.Parse(input.ReadLine());
			_obj.Cost = (float)_obj_Cost;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// Description
			this.Description.Serialize(output);
			// PassiveAlterations
			output.WriteLine(this.PassiveAlterations.Count.ToString());
			for(int PassiveAlterations_it = 0; PassiveAlterations_it < this.PassiveAlterations.Count;PassiveAlterations_it++) {
				this.PassiveAlterations[PassiveAlterations_it].Serialize(output);
			}
			// Cost
			output.WriteLine(((float)this.Cost).ToString());
		}

	}
}
